core/src/main/java/org/apache/calcite/util/TimestampWithTimeZone.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.util;
+
+import java.io.Serializable;
+import java.util.Objects;
+import java.util.TimeZone;
+
+/**
+ * A Timestamp with time zone. It is immutable.
+ */
+public class TimestampWithTimeZone implements Comparable<TimestampWithTimeZone>, Serializable {
+  //~ Instance fields --------------------------------------------------------
+
+  private final long millisecond;
```

Comment:
is this epoch millis or just millis? If it is epoch millis, can you update comments to reflect it?

Reply:
The `millisecond` should be a zone-less TIMESTAMP(the internal representation of `TIMESTAMP WITHOUT TIME ZONE`).  The java doc of the constructor has made it clearly.

Reply:
Can you point to me the class that represents `TIMESTAMP WITHOUT TIME ZONE` in Calcite? I want to double check what type of millis that saves internally?

Reply:
Just a long value. You can see `BuiltInMethod.STRING_TO_TIMESTAMP`